src/Lti/Service/GetUserAssignmentLtiRequestService.php

```diff
@@ -63,13 +63,13 @@ private function checkIfAssignmentCanBeProcessed(Assignment $assignment): void
                 ($maxAttempts === $attemptsCount && $assignment->getState() === Assignment::STATE_COMPLETED)
             )
         ) {
-            throw new AssignmentNotProcessableException(
+            throw new AssignmentUnavailableException(
                 sprintf("Assignment with id '%s' has reached the maximum attempts.", $assignment->getId())
```

Comment:
Don't we need to cast the `$assignment->getId()`

Reply:
Actually it's not necessary, it's automatically casted with the `__toString()` magic method.

Reply:
I understand that, but there are other cases of it being used on printing functions and being cast, so would expect it here or removing from others.

Reply:
Fixed